algorithms/src/snark/marlin/marlin.rs

```diff
@@ -667,27 +717,16 @@ where
         let mut first_comms_consumed = 0;
         let mut first_commitments = batch_sizes
             .iter()
-            .flat_map(|(&circuit_id, &batch_size)| {
+            .flat_map(|(circuit_id, &batch_size)| {
                 let first_comms = comms.witness_commitments[first_comms_consumed..][..batch_size]
                     .iter()
                     .enumerate()
                     .flat_map(|(j, w_comm)| {
-                        [
-                            LabeledCommitment::new_with_info(
-                                &first_round_info[&witness_label(circuit_id, "w", j)],
-                                w_comm.w,
-                            ),
-                            LabeledCommitment::new_with_info(
-                                &first_round_info[&witness_label(circuit_id, "z_a", j)],
-                                w_comm.z_a,
-                            ),
-                            LabeledCommitment::new_with_info(
-                                &first_round_info[&witness_label(circuit_id, "z_b", j)],
-                                w_comm.z_b,
-                            ),
-                        ]
-                    })
-                    .collect_vec();
+                        [LabeledCommitment::new_with_info(
+                            &first_round_info[&witness_label(*circuit_id, "w", j)],
+                            w_comm.w,
+                        )]
+                    });
```

Comment:
We can rewrite this as a `map`:

```suggestion
                   .map(|(j, w_comm)| LabeledCommitment::new_with_info(
                        &first_round_info[&witness_label(*circuit_id, "w", j)],
                        w_comm.w,
                    ));

Reply:
Good catch: 2c5a44a34